swagger.yaml

```diff
@@ -1,75 +1,330 @@
+
+      - name: "types"
+        in: "query"
+        description: "List of types of cards"
+        required: false
+        type: "array"
+        items:
+          type: "string"
+          enum:
+          - "attack"
+          - "skill"
+          - "power"
+          - "status"
+          - "curse"
+        collectionFormat: "multi"
+
+      - name: "name"
+        in: "query"
+        description: "Card name (or part of it)"
+        required: false
+        type: "string"
+
+      - name: "colors"
+        in: "query"
+        description: "List of colors of cards"
+        required: false
+        type: "array"
+        items:
+          type: "string"
+          enum:
+          - "red"
+          - "green"
+          - "blue"
+          - "colorless"
+        collectionFormat: "multi"
+
+      - name: "rarities"
+        in: "query"
+        description: "List of rarities of cards"
+        required: false
+        type: "array"
+        items:
+          type: "string"
+          enum:
+          - "basic"
+          - "common"
+          - "uncommon"
+          - "rare"
+        collectionFormat: "multi"
+
+      - name: "energyMin"
+        in: "query"
+        description: "Minimum value of energy costs of cards (X = 0). Used with energyMax"
+        required: false
+        type: "integer"
+
+      - name: "energyMax"
+        in: "query"
+        description: "Maximum value of energy costs of cards (X = 0). Used with energyMin"
+        required: false
+        type: "integer"
+
+      - name: "keywords"
+        in: "query"
+        description: "List of keywords found in a card's description"
+        required: false
+        type: "array"
+        items:
+          type: "string"
+
+      - name: "number"
+        in: "query"
+        description: "Number of cards to return"
+        required: false
+        type: "integer"
+        default: 10
+
+      - name: "isRandom"
+        in: "query"
+        description: "If true, cards are returned in a random order"
+        required: false
+        type: "boolean"
+        default: true
+
+      responses:
+        200:
+          description: "Card(s) successfully retrieved"
+          schema:
+            $ref: "#/definitions/MultiData"
+        400:
+          description: "Invalid request"
+          schema:
+            $ref: "#/definitions/Error"
+        500:
+          description: "Internal server error"
+          schema:
+            $ref: "#/definitions/Error"
+
+    post:
+      tags:
+      - "cards"
+      summary: "Add a new card to the database"
+      operationId: "addCard"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
       parameters:
-        - $ref: '#/parameters/pretty'
+        - name: "body"
+          in: "body"
+          description: "Card object to be added to the database"
+          required: true
+          schema:
+            $ref: "#/definitions/SingleData"
+
       responses:
         200:
```

Comment:
Perhaps a `201` would be more appropriate when creating a new card object? Sorry if my comment didn't make sense originally.